packages/namekit-react/src/components/DisplayedPrice.tsx

```diff
@@ -0,0 +1,75 @@
+import { type Price, formattedPrice } from "@namehash/ens-utils";
+import React from "react";
+import cc from "classcat";
+
+export const PriceDisplaySize = {
+  Micro: "nk-text-[12px] md:nk-text-[14px] nk-font-normal",
+  Small: "nk-text-[14px] nk-font-semibold",
+  Medium: "nk-text-[20px] nk-font-semibold",
+  Large: "nk-text-[24px] nk-font-bold",
+} as const;
+export type PriceDisplaySize =
+  (typeof PriceDisplaySize)[keyof typeof PriceDisplaySize];
+
+export const CurrencySymbolPosition = {
+  /**
+   * Display the currency symbol to the left of the price. (e.g. $1.00)
+   */
+  Left: "nk-mr-1.5",
+  /**
+   * Display the currency symbol to the right of the price. (e.g. 1.00$)
+   */
+  Right: "nk-ml-1.5",
+} as const;
+export type CurrencySymbolPosition =
+  (typeof CurrencySymbolPosition)[keyof typeof CurrencySymbolPosition];
+
+export interface DisplayedPriceProps {
+  /**
+   * The price to be displayed
+   * @example { currency: Currency.Eth, value: 1000000000000000000n }
+   */
+  price: Price;
+  /**
+   * The currency symbol to display alongside the price.
+   */
+  symbol: React.ReactNode;
```

Comment:
I see from storybook that this param is optional.

Therefore it should be defined with `?` in this interface and we also need to document how it is optional and what happens if it is not defined.

One idea is to make it so that `symbol` is defined as `symbol?: React.ReactNode | null`.

Then, if `symbol` is defined and non-null, we use the defined symbol. Great! This gives full flexibility to people who use `DisplayedPrice`.

If `symbol` is defined and `null` then we can render no symbol at all. This should be a special exception case and we can create a single story for this in storybook. There's no need for so many examples in Storybook to show the idea of hiding the symbol.

If `symbol` is undefined then I'm happy to then have some default definition of the symbol to use. Ex: A `CurrencySymbol` of the appropriate `Currency`, size, and with the tooltip turned on.

It might sound like we're going in circles because I was just making a big push for composability, but please note how we have achieved our composability goals: A fully custom `symbol` can now be provided when desired. Therefore we can have a very simple implementation of a default `symbol` if it is undefined. It's nice to provide defaults that are generally ok. It's important that we don't add any props to `DisplayedPrice` that are for the default `CurrencySymbol` though. If someone wants to customize the `CurrencySymbol` they should pass in a custom value into the `symbol` prop!

Open to feedback on everything above.

Reply:
@lightwalker-eth ok sounds good! i'd only like to hear back a feedback from the getCurrencySymbology placement first, I plan to reuse it here for default use case :)

Reply:
@FrancoAguzzi I continue to believe that `CurrencySymbology` and `getCurrencySymbology` do not belong in `ens-utils`.

Please have a look at the changes that this PR is proposing to `formattedPrice` in `ens-utils`. `formattedPrice` in `ens-utils` worked just fine without `CurrencySymbology` before in the past. I don't see any reason why it should be added now?

In my thinking all changes to `packages/ens-utils/src/price.ts` in this PR should be reverted.

Please let me know if I'm missing something important?

Reply:
@lightwalker-eth it should be good to at least specify `describeCurrencyInTooltip` as in `DisplayedPriceConversion` we want to avoid this tooltip in `DisplayedPrice` rendering:
<img width="96" alt="Captura de Tela 2024-08-29 às 19 41 38" src="https://github.com/user-attachments/assets/15d0f281-5999-4b97-8197-2ae724601b06">